### PR TITLE
Go map type

### PR DESCRIPTION
**Pull request by stefandarius**
## Type of change

-   [x] 🧑‍💻 Improvement

## Description

GO on genezio now supports map types for parameters and struct properties.
Syntax like `map[string]interface{}` (where the value type is any serializable genezio type) is allowed and can be used in the SDK for GO and TS.

## Checklist

-   [x] My code follows the contributor guidelines of this project;
-   [x] New and existing unit tests pass locally with my changes;

### CODE REVIEW

**Review by costinsin — CHANGES_REQUESTED** [review]
The map feature looks good to me. While testing this I have found a bug related to how parameters are generated in the Router.

```go
func (b BackendService) MapType(a map[string]int32) error {
	return nil
}
```

This code fails to compile because `int32` is changed to `int` in the Router code.

This problem can be found when using: int8  int16  int32  int64

Also `uint` (and it's varations uint8 uint16 uint32 uint64 uintptr) are changed to `interface{}`. I think we should fix this and also allow `uint` which will be later used as `number` in TS.

**Review by costinsin — CHANGES_REQUESTED** [review]
```go
func (b BackendService) MapType() error {
	return nil
}
```

This code generates the following ts code:
```ts
static mapType(): Promise<any>;
```

I think the function should return `Promise<void>` instead of `Promise<any>`

File: src/bundlers/go/goBundler.ts
[the remark this answers is not in the document]
Comment (stefandarius) [reply src/bundlers/go/goBundler.ts:1]: I am aware of this issue regarding integer types. Unfortunately the only solution I could find is to declare deserializing strategies for each of those types. I will add a new issue regarding this and the refactorization of the bundlers, but I would like to merge this PR and include the changes in a new one, as those are older issues not related to this PR.

[the remark this answers is not in the document]
Comment (costinsin) [reply src/bundlers/go/goBundler.ts:1]: Ok, this is fine with me

**Review by costinsin — APPROVED** [review]

**Review by costinsin — APPROVED** [review]
tested✅! LGTM

**Pull request merged**